### ISSUE TITLE
Error in code generator

### ISSUE THREAD

**Issue by kehusa** [HEADS-project/heads_ide#87]
In the following code snippet from message receiving:
````
in_ser_sen_readingJson_in: function (msg) {
this.server.receivereadingJsonOnsensor_mqtt(msg.split(';'));
},
in_ser_sma_read_in: function (msg) {
this.server.receivereadOnsmarttracker(msg.split(';'));
},
````

server should have been: .HomeGW_server.

I.e the type is forgotten in the property.

**Comment by Lyadis** [comment]
Could you give the thingml file corresponding?

**Comment by kehusa** [comment]
Sensorjs.thingml in homegw project.



fre. 22. jan. 2016, 17:27 skrev Nicolas Harrand <notifications@github.com>:

**Comment by Lyadis** [comment]
Would this help?

**Comment by kehusa** [comment]
Should have been ServerJs.thingml but I guess you figured it out.

I will try the fix.

Knut Eilif

2016-01-22 17:39 GMT+01:00 Nicolas Harrand <notifications@github.com>:

> Would this help?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/HEADS-project/heads_ide/issues/87#issuecomment-173971923>
> .
>



-- 
| Knut Eilif Husa | Tellu AS | Lensmannslia 4, 1386 Asker - NORWAY| Direct:
+47 452 49496 | www.tellu.no |

This message contains confidential information and is intended only for the
individual named. If you are not the named addressee you should not
disseminate, distribute or copy this e-mail. Please notify the sender
immediately by e-mail if you have received this e-mail by mistake and
delete this e-mail from your system.

**Comment by kehusa** [comment]
Hmm, still the same generated code:
server in the line with
````
this.server.receivereadingJsonOnsensor_mqtt(msg.split(';'));
````
should have been:
````
this.HomeGW_server.receivereadingJsonOnsensor_mqtt(msg.split(';'));
````
````

var AbstractComponent = require('kevoree-entities').AbstractComponent;
var colors = require('colors/safe');
var FallServer = require('./FallServer');
/**
* Kevoree component
* @type {HomeGW}
*/
var HomeGW = AbstractComponent.extend({
toString: 'HomeGW',
//Attributes
dic_HomeGW_server_FallServer_LOC_HYSTERESIS__var : { 
defaultValue: 5},
dic_HomeGW_server_FallServer_FALL_LIMIT__var : { 
defaultValue:  -4},
construct: function() {
this.HomeGW_server = new FallServer("HomeGW_server", null, null, null, null, null, 5,  -4, null, null, false);
this.HomeGW_server.setThis(this.HomeGW_server);
this.HomeGW_server.build();
//Connecting internal ports...
//Connecting ports...
this.HomeGW_server.getReadingonsmarttrackerListeners().push(this.ser_sma_reading_proxy.bind(this));
},

start: function (done) {
this.dictionary.on('HomeGW_server_FallServer_LOC_HYSTERESIS__var', function (newValue) {newValue = Number(newValue);
console.log("Kevoree attribute HomeGW_server_FallServer_LOC_HYSTERESIS__var updated...");
if(this.HomeGW_server.FallServer_LOC_HYSTERESIS__var !== newValue) { console.log("updating ThingML attribute...");
this.HomeGW_server.FallServer_LOC_HYSTERESIS__var = newValue;}});
this.HomeGW_server.FallServer_LOC_HYSTERESIS__var = this.dictionary.getNumber('HomeGW_server_FallServer_LOC_HYSTERESIS__var');this.HomeGW_server.onPropertyChange('LOC_HYSTERESIS', function(newValue) {console.log("ThingML attribute HomeGW_server_FallServer_LOC_HYSTERESIS__var updated...");
if(this.dictionary.getNumber('HomeGW_server_FallServer_LOC_HYSTERESIS__var') !== newValue) {
console.log("updating Kevoree attribute...");
this.submitScript('set '+this.getNodeName()+'.'+this.getName()+'.HomeGW_server_FallServer_LOC_HYSTERESIS__var = "'+newValue+'"');
}}.bind(this));
this.dictionary.on('HomeGW_server_FallServer_FALL_LIMIT__var', function (newValue) {newValue = Number(newValue);
console.log("Kevoree attribute HomeGW_server_FallServer_FALL_LIMIT__var updated...");
if(this.HomeGW_server.FallServer_FALL_LIMIT__var !== newValue) { console.log("updating ThingML attribute...");
this.HomeGW_server.FallServer_FALL_LIMIT__var = newValue;}});
this.HomeGW_server.FallServer_FALL_LIMIT__var = this.dictionary.getNumber('HomeGW_server_FallServer_FALL_LIMIT__var');this.HomeGW_server.onPropertyChange('FALL_LIMIT', function(newValue) {console.log("ThingML attribute HomeGW_server_FallServer_FALL_LIMIT__var updated...");
if(this.dictionary.getNumber('HomeGW_server_FallServer_FALL_LIMIT__var') !== newValue) {
console.log("updating Kevoree attribute...");
this.submitScript('set '+this.getNodeName()+'.'+this.getName()+'.HomeGW_server_FallServer_FALL_LIMIT__var = "'+newValue+'"');
}}.bind(this));
this.HomeGW_server._init();
done();
},

stop: function (done) {
this.HomeGW_server._stop();
done();
},
in_ser_sen_readingJson_in: function (msg) {
this.server.receivereadingJsonOnsensor_mqtt(msg.split(';'));
},
in_ser_sma_read_in: function (msg) {
this.server.receivereadOnsmarttracker(msg.split(';'));
},
ser_sma_reading_proxy: function() {this.out_ser_sma_reading_out(arguments[0]);},
out_ser_sma_reading_out: function(msg) {/* This will be overwritten @runtime by Kevoree JS */}});
````

**Comment by Lyadis** [comment]
Oh, but HomeGW is the name of your configuration isn't it? (Not the type of your instance?)

**Comment by brice-morin** [comment]
I think I see the problem. Will fix it.

**Comment by kehusa** [comment]
Sorry about mentioning the type. It was the configuration name that was
missing

2016-01-22 18:50 GMT+01:00 Brice Morin <notifications@github.com>:

> I think I see the problem. Will fix it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/HEADS-project/heads_ide/issues/87#issuecomment-173991283>
> .
>



-- 
| Knut Eilif Husa | Tellu AS | Lensmannslia 4, 1386 Asker - NORWAY| Direct:
+47 452 49496 | www.tellu.no |

This message contains confidential information and is intended only for the
individual named. If you are not the named addressee you should not
disseminate, distribute or copy this e-mail. Please notify the sender
immediately by e-mail if you have received this e-mail by mistake and
delete this e-mail from your system.

**Comment by brice-morin** [comment]
It seems @Lyadis fixed the bug. I regenerated the Eclipse plugins and also added some comments in the generated code to make future (hopefully none) debugging easier and also explain the mapping of ports between ThingML and Kevoree.

I suspect a bug in ThingML, which is really not helping writing this KevoreeJS compiler right. It seems than sometimes `myInstance.getName()` just returns the instance name, but sometimes it returns the same as `myCfg.getName() + "_" + myInstance.getName()`, which probably explains why we added `myCfg.getName() + "_"` in some places and not in some other places (actually, we probably do not need the configuration name, but probably added it at some point as a work around). We should investigate the problem in ThingML and then refactor the KevoreeJS compiler not to make it more tricky than it is really.
